### PR TITLE
Add workflow events documentation

### PR DESCRIPTION
**Pull request by grena**

### CODE REVIEW

File: reference/events/workflow_events.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+=========================================
+
+Workflow events are dispatched in the PIM on some workflow actions, mainly about product draft steps (*sent for approval, rejected, deleted, approved...*).
+They are basically useful if you want to add your business logic before or after those specific actions.
+
+
+``pimee_workflow.product_draft.post_approve``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched after a product draft proposal is approved.
```
Comment (rybus) [inline reference/events/workflow_events.rst]: is being approved

File: reference/events/workflow_events.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+Workflow events are dispatched in the PIM on some workflow actions, mainly about product draft steps (*sent for approval, rejected, deleted, approved...*).
+They are basically useful if you want to add your business logic before or after those specific actions.
+
+
+``pimee_workflow.product_draft.post_approve``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Event Class: `GenericEvent <http://api.symfony.com/2.7/Symfony/Component/EventDispatcher/GenericEvent.html>`_
+
+This event is dispatched after a product draft proposal is approved.
```
Comment (rybus) [inline reference/events/workflow_events.rst]: has been approved

File: reference/events/workflow_events.rst [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+===============================  ===============
+
+.. _RefuseNotificationSubscriber: https://github.com/akeneo/pim-enterprise-dev/blob/master/src/PimEnterprise/Bundle/WorkflowBundle/EventSubscriber/ProductDraft/RefuseNotificationSubscriber.php
+
+``pimee_workflow.product_draft.post_remove``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Event Class: `PublishedProductEvent <https://github.com/akeneo/pim-enterprise-dev/blob/master/src/PimEnterprise/Bundle/WorkflowBundle/Event/PublishedProductEvent.php>`_
+
+This event is dispatched after a product draft proposal is removed.
```
Comment (rybus) [inline reference/events/workflow_events.rst]: has been removed

File: reference/events/workflow_events.rst [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+.. _RemoveNotificationSubscriber: https://github.com/akeneo/pim-enterprise-dev/blob/master/src/PimEnterprise/Bundle/WorkflowBundle/EventSubscriber/ProductDraft/RemoveNotificationSubscriber.php
+
+
+``pimee_workflow.published_product.post_publish``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Event Class: `PublishedProductEvent <https://github.com/akeneo/pim-enterprise-dev/blob/master/src/PimEnterprise/Bundle/WorkflowBundle/Event/PublishedProductEvent.php>`_
+
+This event is dispatched after a product is published.
```
Comment (rybus) [inline reference/events/workflow_events.rst]: has been published

File: reference/events/workflow_events.rst [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+.. _DetachProductPostPublishSubscriber: https://github.com/akeneo/pim-enterprise-dev/blob/master/src/PimEnterprise/Bundle/WorkflowBundle/EventSubscriber/PublishedProduct/DetachProductPostPublishSubscriber.php
+
+
+``pimee_workflow.product_draft.pre_ready``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Event Class: `GenericEvent <http://api.symfony.com/2.7/Symfony/Component/EventDispatcher/GenericEvent.html>`_
+
+This event is dispatched after a product is published.
```
Comment (rybus) [inline reference/events/workflow_events.rst]: has been published

File: reference/events/workflow_events.rst [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+
+``pimee_workflow.product_draft.pre_apply``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Event Class: `GenericEvent <http://api.symfony.com/2.7/Symfony/Component/EventDispatcher/GenericEvent.html>`_
+
+This event is dispatched after a product is published.
```
Comment (rybus) [inline reference/events/workflow_events.rst]: has been published

File: reference/events/workflow_events.rst [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+
+``pimee_workflow.product_draft.pre_approve``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Event Class: `GenericEvent <http://api.symfony.com/2.7/Symfony/Component/EventDispatcher/GenericEvent.html>`_
+
+This event is dispatched after a product is published.
```
Comment (rybus) [inline reference/events/workflow_events.rst]: has been published

File: reference/events/workflow_events.rst [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+
+
+``pimee_workflow.product_draft.pre_refuse``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Event Class: `GenericEvent <http://api.symfony.com/2.7/Symfony/Component/EventDispatcher/GenericEvent.html>`_
+
+This event is dispatched after a product is published.
```
Comment (rybus) [inline reference/events/workflow_events.rst]: has been published

File: reference/events/workflow_events.rst [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+
+
+``pimee_workflow.product_draft.pre_remove``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Event Class: `GenericEvent <http://api.symfony.com/2.7/Symfony/Component/EventDispatcher/GenericEvent.html>`_
+
+This event is dispatched after a product is published.
```
Comment (rybus) [inline reference/events/workflow_events.rst]: has been published

**Comment by ClementGautier** [comment]
GTM for me, even if I would have prefer that the product draft remove event was a generic storage remove event.

File: reference/events/workflow_events.rst [hunk 10]
```diff
@@ -0,0 +1,237 @@
+Workflow events (Enterprise Edition only)
+=========================================
+
+Workflow events are dispatched in the PIM on some workflow actions, mainly about product draft steps (*sent for approval, rejected, deleted, approved...*) and publish feature.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: mainly in product... and publication feature.

File: reference/events/workflow_events.rst [hunk 11]
```diff
@@ -0,0 +1,237 @@
...
+
+When a product is unpublished:
+    1) ``pimee_workflow.published_product.pre_unpublish``
+    2) The product object is unpublished
+    3) ``pimee_workflow.published_product.post_unpublish`` (**only on flush!**)
+
+``pimee_workflow.published_product.pre_publish``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched **before** a product has been published.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is published

File: reference/events/workflow_events.rst [hunk 12]
```diff
@@ -0,0 +1,237 @@
...
+
+**Event Class**: ``PimEnterprise\Bundle\WorkflowBundle\Event\PublishedProductEvent``
+
+``pimee_workflow.published_product.post_publish``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched **after** a product has been published.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is published

File: reference/events/workflow_events.rst [hunk 13]
```diff
@@ -0,0 +1,237 @@
...
+===========================================================================================================  ===============
+Listener Class Name                                                                                          Priority
+===========================================================================================================  ===============
+``PimEnterprise\Bundle\WorkflowBundle\EventSubscriber\PublishedProduct\DetachProductPostPublishSubscriber``  0
+===========================================================================================================  ===============
+
+``pimee_workflow.published_product.pre_unpublish``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched **before** a product has been unpublished.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is unpublished

File: reference/events/workflow_events.rst [hunk 14]
```diff
@@ -0,0 +1,237 @@
...
+
+When a product draft is deleted:
+    1) ``pimee_workflow.product_draft.pre_remove``
+    2) Product draft is **removed from the DB**
+    3) ``pimee_workflow.product_draft.post_remove``
+
+``pimee_workflow.product_draft.pre_ready``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched **before** a product draft status has been set to "ready".
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is set to

File: reference/events/workflow_events.rst [hunk 15]
```diff
@@ -0,0 +1,237 @@
...
+==============================================================================================  ===============
+Listener Class Name                                                                             Priority
+==============================================================================================  ===============
+``PimEnterprise\Bundle\WorkflowBundle\EventSubscriber\ProductDraft\SendForApprovalSubscriber``  0
+==============================================================================================  ===============
+
+``pimee_workflow.product_draft.pre_approve``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched **before** a product draft has been approved.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is approved

File: reference/events/workflow_events.rst [hunk 16]
```diff
@@ -0,0 +1,237 @@
...
+
+**Event Class**: `GenericEvent <http://api.symfony.com/2.7/Symfony/Component/EventDispatcher/GenericEvent.html>`_
+
+``pimee_workflow.product_draft.post_approve``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched **after** a product draft has been approved.
+The product **has been updated and saved** with the new values, and the product draft has been removed or updated.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: ** is updated..**... draft is removed or updated

File: reference/events/workflow_events.rst [hunk 17]
```diff
@@ -0,0 +1,237 @@
...
+==================================================================================================  ===============
+Listener Class Name                                                                                 Priority
+==================================================================================================  ===============
+``PimEnterprise\Bundle\WorkflowBundle\EventSubscriber\ProductDraft\ApproveNotificationSubscriber``  0
+==================================================================================================  ===============
+
+``pimee_workflow.product_draft.pre_partial_approve``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched **before** a product draft has been **partially** approved.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is **partially**

File: reference/events/workflow_events.rst [hunk 18]
```diff
@@ -0,0 +1,237 @@
...
+A partial approve could be about a specific attributes/locale/channel of the product draft.
+
+**Event Class**: `GenericEvent <http://api.symfony.com/2.7/Symfony/Component/EventDispatcher/GenericEvent.html>`_
+
+``pimee_workflow.product_draft.post_partial_approve``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched **before** a product draft has been **partially** approved.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is partially

File: reference/events/workflow_events.rst [hunk 19]
```diff
@@ -0,0 +1,237 @@
...
+The product **has been updated and saved** with the new values, and the product draft has been removed or updated.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is updated... draft is removed

File: reference/events/workflow_events.rst [hunk 20]
```diff
@@ -0,0 +1,237 @@
...
+==================================================================================================  ===============
+Listener Class Name                                                                                 Priority
+==================================================================================================  ===============
+``PimEnterprise\Bundle\WorkflowBundle\EventSubscriber\ProductDraft\ApproveNotificationSubscriber``  0
+==================================================================================================  ===============
+
+``pimee_workflow.product_draft.pre_apply``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched **before** a product object has been updated from draft values.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is updated

File: reference/events/workflow_events.rst [hunk 21]
```diff
@@ -0,0 +1,237 @@
...
+
+This event is dispatched **after** a product object has been updated from draft values.
+Note that **the product has not been saved yet**, only the product object has been updated.
+
+**Event Class**: `GenericEvent <http://api.symfony.com/2.7/Symfony/Component/EventDispatcher/GenericEvent.html>`_
+
+``pimee_workflow.product_draft.pre_refuse``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched **before** a product draft has been refused.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is refused

File: reference/events/workflow_events.rst [hunk 22]
```diff
@@ -0,0 +1,237 @@
...
+=================================================================================================  ===============
+Listener Class Name                                                                                Priority
+=================================================================================================  ===============
+``PimEnterprise\Bundle\WorkflowBundle\EventSubscriber\ProductDraft\RefuseNotificationSubscriber``  0
+=================================================================================================  ===============
+
+``pimee_workflow.product_draft.pre_partial_refuse``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched **before** a product draft has been **partially** refused.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is **partially** refused

File: reference/events/workflow_events.rst [hunk 23]
```diff
@@ -0,0 +1,237 @@
...
+A partial refuse could be about a specific attributes/locale/channel of the product draft.
+
+**Event Class**: `GenericEvent <http://api.symfony.com/2.7/Symfony/Component/EventDispatcher/GenericEvent.html>`_
+
+``pimee_workflow.product_draft.post_partial_refuse``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched **before** a product draft has been **partially** refused.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is partially refused

File: reference/events/workflow_events.rst [hunk 24]
```diff
@@ -0,0 +1,237 @@
...
+The product draft has been removed or updated.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is removed

File: reference/events/workflow_events.rst [hunk 25]
```diff
@@ -0,0 +1,237 @@
...
+=================================================================================================  ===============
+Listener Class Name                                                                                Priority
+=================================================================================================  ===============
+``PimEnterprise\Bundle\WorkflowBundle\EventSubscriber\ProductDraft\RefuseNotificationSubscriber``  0
+=================================================================================================  ===============
+
+``pimee_workflow.product_draft.pre_remove``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This event is dispatched **before** a product draft has been removed.
```
Comment (MarieMinasyan) [inline reference/events/workflow_events.rst]: is removed

**Comment by rybus** [comment]
haha @MarieMinasyan you commented the exact opposite as I did :p Sorry @grena for the wrong comments and thanks @MarieMinasyan for the explanations!

**Comment by grena** [comment]
Thank you :)

**Pull request closed without merging**